src/py42/services/savedsearch.py

```diff
@@ -61,3 +62,20 @@ def execute(self, search_id, page_number=None, page_size=None):
         """
         query = self.get_query(search_id, page_number=page_number, page_size=page_size)
         return self._file_event_client.search(query)
+
+    def execute_get_all(self, search_id, page_token=""):
+        """
+        Executes a saved search for given search Id, returns a page of events with a token in the response to retrieve next page.
+
+        Args:
+            search_id (str): Unique search Id of the saved search.
+            page_token (str, optional): A token used to indicate the starting point for
+                additional page results. For the first page, do not pass ``page_token``. For
+                all consecutive pages, pass the token from the previous response from
+                field ``nextPgToken``. Defaults to empty string.
+        Returns:
+            :class:`py42.response.Py42Response`: A response containing a page of events.
+        """
+        query = self.get_query(search_id)
```

Comment:
on thought I just had -- it is unfortunate that that if you are paging through this that we will have to make the api call to get the query every time. I wonder if there's a good way could design that that would allow us to avoid that overhead.

Reply:
I imagine this is why in fileevents querying you pass the `Query` instead of the `search_id`?  If we wanted this one to work similarly, we could switch it to take that parameter instead - however then it would diverge from the original 'execute()' method. 

Or perhaps if we wanted to keep the param as `search_id` we could have the function make this api call and then call some helper function on that returned Query?  I'm not sure how closely we want this new method to mirror `execute()`.

Reply:
If we didn't make the user handle the page tokens (but instead handled that internally and just yielded pages like other `get_all` methods), we could avoid the overhead, but since "paging" requires multiple manual calls of this same method I don't know if we'd be able to safely cache the query results, since we don't know when the user is "done" in order to clear the cache. 

Any reason not to do that here? I know we're not consistent with the regular `securitydata.search_all_file_events` in how we handle page tokens, but the end user doesn't really need to know that saved search is doing the same thing automatically for them, right?

Reply:
So sounds like we just want to leave it as is to stay consistent with the other `search_all_file_events()` method for now, and then we'll plan on improving their design in the future

Reply:
I kind of realized that we can achieve what we want already by doing the following:
- call `get_query`
- pass the result of that to `securitydata.search_all_file_events`, use that method as normal

Adding a `search_all_file_events` method here would encourage users to use something that results in making extra, unneeded calls -- maybe we can just update the docs of the `search_file_events` and/or `get_query` method here to call out how to go about doing this, and get rid of `search_all_file_events`?

Reply:
Hmm, I like that idea..  make the "main" way to execute a saved search be to get the query object, then use it in the normal file event search methods. 

Maybe we could mark `.execute` as deprecated and update documentation to highlight the better way?

Reply:
So there would be `securitydata.search_all_file_events()` which takes a query and an optional page token and returns a response _with_ page tokens (if applicable).
There'd be securitydata.search_file_events() which takes a query and returns a response _without_ page tokens.

We'd remove `savedsearch.search_all_file_events()` entirely.
And then we'd have savedsearch.search_file_events() which takes a search id and returns a response without page tokens (currently be calling `execute()` which also calls `get_query()`.  Should we remove this method as well and plan on documenting that users simply use `get_query()` and then pass that into the existing `securitydata` file event search methods?

Reply:
>  Should we remove this method as well and plan on documenting that users simply use get_query() and then pass that into the existing securitydata file event search methods?

No, `execute` already exists and this is just a better name for it -- and removing it would be breaking.
There's still value to getting the first 10K results using just a search id, imo.